bark_ml/observers/graph_observer.py

```diff
@@ -0,0 +1,554 @@
+# Copyright (c) 2020 fortiss GmbH
+#
+# Authors: Marco Oliva, Silvan Wimmer
+#
+# This work is licensed under the terms of the MIT license.
+# For a copy, see <https://opensource.org/licenses/MIT>.
+
+import numpy as np
+import logging
+import tensorflow as tf
+from gym import spaces
+from typing import Dict
+from collections import OrderedDict
+
+from bark.core.models.dynamic import StateDefinition
+from bark.core.geometry import Distance, Point2d
+from bark.runtime.commons.parameters import ParameterServer
+
+from bark_ml.observers.observer import StateObserver
+
+class GraphObserver(StateObserver):
+  """
+  This observer converts an `ObservsedWorld` instance into
+  a graph-structured observation, consisting of nodes (with
+  features), an adjacency matrix expressing the (directed)
+  connections between nodes in the graph, and edges (with
+  features), expressing the relative information between
+  the agents that constitute these connections.
+  """
+
+  def __init__(self,
+               normalize_observations=True,
```

Comment:
Everything else is configured using the `ParameterServer`, should this not also be the case for the normalization?
Would make it easier to store all parameters.

Reply:
Good point. I will change this.